src/backend/common/UsageTracker.h

```diff
@@ -0,0 +1,81 @@
+// Copyright 2017 The NXT Authors
```

Comment:
Currently the usage tracking is the same for textures and buffers and this factorization makes sense, but textures will eventually have per-mip tracking, so this class will have to be split again. When that happens I think it will make sense to fold the UsageTracker in the resource class again.